JavascriptWebAppSample/index.html

```diff
@@ -0,0 +1,90 @@
+            <a href="#" onclick="authContext.logOut(); return false;">Log out</a>
+        </p>
+        <p id="username"></p>
+        <pre id="api_response"></pre>
+
+        <script type="text/javascript">
+            //========= Input Vars ================================
+            var clientId = '57c5c51c-2036-433a-befa-ebefc7f10689'; //Update with your app regirstion's Application ID
+            var replyUri = 'http://localhost:8080'; //Where you will receive the token. 
+            var logoutRedirectUri = 'http://localhost:8080'; //Update if you are hosting your webpage somewhere else
+            var vstsApi = 'https://myaccount.visualstudio.com/_apis/projects?api-version=2.0';//'https://app.vssps.visualstudio.com/_apis/profile/profiles/me?api-version=1.0';//Update if want to access a different vsts API
+            var vstsResourceId = '499b84ac-1321-427f-aa17-267ca6975798'; //Do not change
+            //=====================================================
+            
+            // Set up ADAL
+            var authContext = new AuthenticationContext({
+                clientId: clientId,
+                redirectUri:replyUri,
+                postLogoutRedirectUri: logoutRedirectUri //Where you return upon signout
+            });
+            // Make an AJAX request to the VSTS REST API and print the response as JSON.
+            var getCurrentUserInfo = function (access_token) {
+                document.getElementById('api_response').textContent = 'Calling API...';
+                var xhr = new XMLHttpRequest();
+                //API called with ADAL token
+                xhr.open('GET', vstsApi, true);
+                xhr.setRequestHeader('Authorization', 'Bearer ' + access_token);
+                xhr.onreadystatechange = function () {
+                    if (xhr.readyState === 4 && xhr.status === 200) {
+                        // Parse Successful Response
+                        document.getElementById('api_response').textContent =
+                            JSON.stringify(JSON.parse(xhr.responseText), null, '  ');
+                    } else {
+                        // Handle Error
+                        document.getElementById('api_response').textContent =
+                            'ERROR:\n\n' + xhr.responseText;
+                    }
+                };
+                xhr.send();
+            }
+            if (authContext.isCallback(window.location.hash)) {
+                // Handle redirect after token requests
+                authContext.handleWindowCallback();
+                var err = authContext.getLoginError();
+                if (err) {
+                    // Handle error
+                    document.getElementById('api_response').textContent =
+                        'ERROR:\n\n' + err;
+                }
+            } else {
+                // If logged in, get access token and make an API request
+                var user = authContext.getCachedUser();
```

Comment:
If the user has already authenticated, on "authcontext.login()" instead of launching an interactive login prompt it will user credentials to get an access token in the background. 

In short all of this is taken care of by the the ADAL JS library.